src/conn/conn_api.c

```diff
@@ -1048,6 +1048,69 @@ __conn_load_extensions(WT_SESSION_IMPL *session, const char *cfg[], bool early_l
     return (ret);
 }
 
+/*
+ * __conn_add_target_uri --
+ *     Add the target uri to the backup uri hash table.
+ */
+static int
+__conn_add_target_uri(WT_SESSION_IMPL *session, const char *name, size_t len)
+{
+    WT_BKUP_TARGET *new_target_uri;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    uint64_t bucket, hash;
+
+    conn = S2C(session);
+
+    WT_ERR(__wt_calloc_one(session, &new_target_uri));
+    WT_ERR(__wt_strndup(session, name, len, &new_target_uri->name));
+
+    hash = __wt_hash_city64(name, len);
+    bucket = hash & (conn->hash_size - 1);
+    new_target_uri->name_hash = hash;
+    WT_BKUP_TARGET_INSERT(conn, new_target_uri, bucket);
```

Comment:
Why did you decide to wrap this in a macro? It's only used here, so I'd have thought the insert along with a comment would be adequate.